### PR TITLE
Nightlies

### PR DESCRIPTION
**Pull request by mgasner**
#412

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -65,8 +65,15 @@ workflows:
             - airline-demo-py35
             - airline-demo-py27
           context: coveralls
-
-
+  nightlies:
+    schedule:
+      cron: "0 8 * * *"
```
Comment (schrockn) [inline .circleci/config.yml]: !!! no idea circleci could this 👍

File: .circleci/config.yml [hunk 2]
```diff
@@ -422,4 +429,27 @@ jobs:
...
+
+      - run:
+          name: Install requitements
+          command: sudo pip install -r bin/requirements.txt
+
+          - run:
+          name: Publish nightlies
+          command: python bin/publish.py --nightlies
+
+      - run:
```
Comment (schrockn) [inline .circleci/config.yml]: format on a cron? hmmm

File: .circleci/images/build.sh [hunk 3]
```diff
@@ -8,5 +8,4 @@ docker build -f Dockerfile-3.7.2 -t dagster/cci-test:3.7.2 .
 docker push dagster/cci-test:2.7
 docker push dagster/cci-test:3.5
 docker push dagster/cci-test:3.6
-# Pending resolution of creds issue
-# docker push dagster/cci-test:3.7.2
+docker push dagster/cci-test:3.7.2
```
Comment (schrockn) [inline .circleci/images/build.sh]: danke

File: python_modules/dagit/setup.py [hunk 4]
```diff
@@ -20,66 +21,80 @@ def long_description():
...
-        # 'rope>=0.10.7',
-        # 'Sphinx>=1.7.5',
-        # 'sphinx-autobuild>=0.7.1',
-        # 'yapf>=0.22.0',
-        # 'twine>=1.11.0',
-        # 'pre-commit'>=1.10.1',
-    ],
-    entry_points={"console_scripts": ['dagit = dagit.cli:main']},
-)
+parser = argparse.ArgumentParser()
```
Comment (schrockn) [inline python_modules/dagit/setup.py]: we might want to consider a shared library to eliminate code duplication here. maybe make an issue to track

**Comment by mgasner** [comment]
I love the idea of a shared lib but it can't be inside dagster because of
the circular dependency -- setup.py is outside the package (and is called
to install the package).

**Comment by mgasner** [comment]
This is cruft and is going away

**Comment by schrockn** [comment]
ah so we would have to have our own setup tools module just for that

Sent via Superhuman ( https://sprh.mn/?vip=schrockn@elementl.com )

**Comment by mgasner** [comment]
it’s not insane, managing this stuff is always such a headache

**Pull request merged**